wait/http.go

```diff
@@ -142,24 +142,50 @@ func (ws *HTTPStrategy) WaitUntilReady(ctx context.Context, target StrategyTarge
 		return
 	}
 
-	var port nat.Port
-	port, err = target.MappedPort(ctx, ws.Port)
-
-	for port == "" {
-		select {
-		case <-ctx.Done():
-			return fmt.Errorf("%s:%w", ctx.Err(), err)
-		case <-time.After(ws.PollInterval):
-			if err := checkTarget(ctx, target); err != nil {
-				return err
+	var mappedPort nat.Port
+	if ws.Port == "" {
+		ports, err := target.Ports(ctx)
+		for err != nil {
+			select {
+			case <-ctx.Done():
+				return fmt.Errorf("%s:%w", ctx.Err(), err)
+			case <-time.After(ws.PollInterval):
+				if err := checkTarget(ctx, target); err != nil {
+					return err
+				}
+
+				ports, err = target.Ports(ctx)
+			}
+		}
+		for k, bindings := range ports {
```

Comment:
Let's create a visual separation of concerns here
```suggestion
		}

		for k, bindings := range ports {
```